### PR TITLE
New Attribute i

### PR DESCRIPTION
**Pull request by hotstepper13**
Allows connecting to an IP instead of dns to check multiple servers
behind loadbalancer or reverse proxy with their own certificate

### CODE REVIEW

File: check_ssl.sh [hunk 1]
```diff
@@ -47,6 +48,10 @@ while getopts :H:p:P:w:c:h FLAG; do
 			CRITICAL_DAYS=$OPTARG
 			;;
 
+    i) #set ip to connect (instead of dns)
```
Comment (karras) [inline check_ssl.sh:51]: Please align this case with the other ones.

File: check_ssl.sh [hunk 2]
```diff
@@ -84,14 +93,14 @@ DATE_ACTUALLY_SECONDS=$(date +"%s")
...
-	HOST_CHECK=$(openssl s_client -servername "${HOST}" -connect "${HOST}":"${PORT}" 2>&- | openssl x509 -enddate -noout)
+	HOST_CHECK=$(openssl s_client -servername "${HOST}" -connect "${IP}":"${PORT}" 2>&- | openssl x509 -enddate -noout)
 	while [ "${?}" = "1" ]; do
 		echo "Check Hostname"
 		exit 1
 	done
 	DATE_EXPIRE_SECONDS=$(echo "${HOST_CHECK}" | sed 's/^notAfter=//g' | xargs -I{} date -d {} +%s)
 else
-		HOST_CHECK=$(openssl s_client -servername "${HOST}" -connect "${HOST}":"${PORT}" -starttls "${PROTOCOL}" 2>&- | openssl x509 -enddate -noout)
+		HOST_CHECK=$(openssl s_client -servername "${HOST}" -connect "${IP}":"${PORT}" -starttls "${PROTOCOL}" 2>&- | openssl x509 -enddate -noout)
```
Comment (karras) [inline check_ssl.sh:103]: This line should be aligned as well. :)

**Review by karras — CHANGES_REQUESTED** [review]
Thanks for the PR, looks great! Could you please fix the alignments? Then I'll merge it.

**Comment by hotstepper13** [comment]
Sorry not sure what to do. On my System(s) (windows and Mac with eclipse) the indenting looks good. Feel free to change it in order to match your requirements. Most probably this may because lacking git skills on my side to I apologize for any inconvenience.

**Comment by luk43** [comment]
Hi @hotstepper13 
You can fill in the ip address of the server just like you would for the hostname.
It works the same way. No changes needed.

**Comment by hotstepper13** [comment]
One hint: If you validate a certificate by common name e.g. www.example.com but let the check run from localhost, it might make sense to connect with ip 127.0.0.1 instead of a regular dns entry.

Especially if you have (for instance) 3 Servers behind a (L4) loadbalancer and you want to make sure that each server has the correct certificate installed. Then you let the check run on each server instead of the dns record.

That was the intention for the change and for this use case the change WAS needed :)

**Comment by luk43** [comment]
merged now through [this commit](https://github.com/adfinis-sygroup/check-ssl/commit/172faae296879c5003584e78490b265bbdf92b7a).
Thanks @hotstepper13 for explaining the problem one more time. ^^
cheers

**Pull request closed without merging**